### ISSUE TITLE
Add Basic Authentication (Frontend)

### ISSUE THREAD

**Issue by brandon-dore** [brandon-dore/rank-tracker#4]
**blocked by issue #3 **
After auth has been added to the backend, the UI will now have to allow a user to perform these authentication functions. This includes:

- Sign Up Page
- Sign In Page/Modal
- Persisting the users login using JWT ([example](https://jasonwatmore.com/post/2022/06/15/react-18-redux-jwt-authentication-example-tutorial))
- Logout Button

**Comment by brandon-dore** [comment]
> Are we implementing RBAC here also? Depending on when we add the admin dashboard it'd be nice to create an `AuthenticationGuard` component that locks it down to only admins. 

Yep I reckon this is in the scope of the ticket since we already have a roles table in the DB. I found a good example of express middleware for this [here](https://code.pieces.app/blog/role-based-access-systems-in-nodejs). Then we are blocking reqs at the API level using RBAC.

**Comment by AmosV-dev** [comment]
Are we implementing RBAC here also? Depending on when we add the admin dashboard it'd be nice to create an AuthenticationGuard component that locks it down to only admins.

**Comment by brandon-dore** [comment]
> Are we implementing RBAC here also? Depending on when we add the admin dashboard it'd be nice to create an AuthenticationGuard component that locks it down to only admins.

Yep I reckon this is in the scope of the ticket since we already have a roles table in the DB. I found a good example of express middleware for this [here](https://code.pieces.app/blog/role-based-access-systems-in-nodejs). Then we are blocking reqs at the API level using RBAC.
